test: Add integration tests to check all kubebuilder:validation tags

viccuad opened the pull request:
## Description


Add 2 tests that cover all current kubebuilder:validation tags.

The Kubebuilder tags are on the types:

`PolicySpec{}`: shared by AdmissionPolicies and ClusterAdmissionPolicies
`PolicyGroupmember{}`, `GroupSpec{}`: shared by AdmissionPolicyGroups and ClusterAdmissionPolicyGroups

Hence create 1 testcase per namespaced and clusterwide policy type.

## Test

<!-- Please provides a short description about how to test your pullrequest -->
To test this pull request, you can run the following commands:

<!--
```shell
cp <to_package_directory>
go test
```
-->

## Additional Information

### Tradeoff

<!-- Please describe, if any, the tradeoffs that you found acceptable in this pull request -->

### Potential improvement

<!-- Please describe, if any, potential improvement that you are envisioning -->

jvanz reviewed (changes requested):
I have a question about how you wrote the test.

viccuad commented on internal/controller/clusteradmissionpolicy_controller_test.go:
```diff
@@ -33,6 +33,33 @@ import (
+		var policyName string
+		var policy *policiesv1.ClusterAdmissionPolicy
+
+		BeforeAll(func() {
```
Good point, this should indeed by a It(). Refactored, and separated the cases further (so we actually test each of the cases instead of all of them failing at once).

viccuad commented:
Ready for review. This is a doozy.

Removed the test cases verifying that `spec.module` is required. That they are required is working as expected, checked the code, checked the resulting crds (see [here](https://github.com/kubewarden/kubewarden-controller/blob/2e6d8e39b61da0cea2627d12dbaac522da981af6/config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml#L489)).

The thing is, this kubebuilder/controller-gen markers (`+required` or  `+kubebuilder:validation:Required`) have no effect. The `spec.module` field is set as required, as it doesn't have any json tags such as `omitempty`.

I have checked the code, checked the testcases, verified that the testcases TestEnv correctly consumes the CRDs definition, verified that we are running these specific tests as `in-cluster` so the k8s api is the one verifying the CR instantiation, etc. Yet I cannot trigger the testcase and have it correctly fail when a `spec.module` is not passed.

Hence, I have removed these `spec.module` tests are left the others, which correctly test the kubebuilder markers.

In any case, since we check the CRDs YAMLs into git, if we were to mistag the `spec.module` we would realize then.

viccuad commented:
Thanks! Rebased and dropped those unneeded functions, plus reworded the commit message as its scope was reduced.
